tests/unit/browser/test_commands.py

```diff
@@ -0,0 +1,212 @@
+        )
+
+    @pytest.mark.parametrize("count, index, expect", [
+        (8, None, "8"),
+        (8, "0", "8"),
+        (8, "2", "8"),
+        (0, "2", "0"),
+        (None, "2", "2"),
+    ])
+    @mock.patch.object(commands.CommandDispatcher, '_resolve_buffer_index')
+    def test_runs_with_count_or_index(self, resolve_mock, tabbed_browser,
+                                      count, index, expect):
+        """Accept an index arg or a count, prefer count."""
+        resolve_mock.return_value = (tabbed_browser, None)
+        uut = commands.CommandDispatcher(0, tabbed_browser)
+        uut.buffer(count=count, index=index)
+        resolve_mock.assert_called_once_with(expect)
+
+    @mock.patch.object(commands.CommandDispatcher, '_resolve_buffer_index')
+    def test_window_raised(self, resolve_mock, tabbed_browser):
+        """Test the resolved window and tab are presented to the user."""
+        tab = mock.Mock()
+        resolve_mock.return_value = (tabbed_browser, tab)
+        uut = commands.CommandDispatcher(0, tabbed_browser)
+        uut.buffer(0)
+
+        # Raise the window, probably fragile.
+        window = tabbed_browser.widget.window.return_value
+        window.activateWindow.assert_called_once_with()
+        window.raise_.assert_called_once_with()
+
+        tabbed_browser.widget.setCurrentWidget.assert_called_once_with(tab)
+
+    def test_tab_resolves(self, tabbed_browser_stubs):
+        """Test the happy path of getting a fully specified index."""
+        fake_tab = mock.Mock()
+        tabbed_browser_stubs[0].widget.tabs = [
+            fake_tab
+        ]
+        uut = commands.CommandDispatcher(0, None)
+        tabbed_browser, tab = uut._resolve_buffer_index('0/1')
+        assert tabbed_browser is tabbed_browser_stubs[0]
+        assert tab is fake_tab
+
+    def test_tab_resolves_with_no_winid(
+            self, tabbed_browser_stubs, app, monkeypatch
+    ):
+        """Test resolving and index with no window specified."""
+        fake_tab = mock.Mock()
+        tabbed_browser_stubs[1].widget.tabs = [
+            fake_tab
+        ]
+        window = mock.Mock()
+        window.return_value.win_id = 1
+        monkeypatch.setattr(app, 'activeWindow', window)
+
+        uut = commands.CommandDispatcher(0, None)
+        tabbed_browser, tab = uut._resolve_buffer_index('1')
+        assert tabbed_browser is tabbed_browser_stubs[1]
+        assert tab is fake_tab
+
+    def test_tab_resolving_no_active_window(self, app, monkeypatch):
+        """Ensure a failure to lookup the active window is handled."""
+        window = mock.Mock()
+        window.return_value = None
+        monkeypatch.setattr(app, 'activeWindow', window)
+
+        uut = commands.CommandDispatcher(0, None)
+        with pytest.raises(cmdexc.CommandError) as err:
+            uut._resolve_buffer_index('1')
+        assert str(err).endswith(
+            "No window specified and couldn't find active window!"
+        )
```

Comment:
You can use `pytest.raises(cmdexc.CommandError, match="No window specified and couldn't find active window!")` instead.

(same for the other tests below)